src/main/java/cubicchunks/api/ICubicWorldGenerator.java

```diff
@@ -0,0 +1,48 @@
+/*
+ *  This file is part of Cubic Chunks Mod, licensed under the MIT License (MIT).
+ *
+ *  Copyright (c) 2015 contributors
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in
+ *  all copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *  THE SOFTWARE.
+ */
+package cubicchunks.api;
+
+import java.util.Random;
+
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+/**
+ * Implement this interface to your world generators and register them in
+ * {@link CubeGeneratorsRegistry}.
+ */
+public interface ICubicWorldGenerator {
```

Comment:
This comment is work in progress, but I don't have much time now, so I will leave it. I realized there are problems with this approach.

I would suggest making ICubicWorldGenerator extend the forge IWorldGenerator (is that the name?) and use WorldGenerators registered with Forge.

This would allow other kods to do something like this:

```java
ModWorldGenerator implements IWorldGenerator
ModWorldGeneratorCubicChunks extends ModWorldGenerator implements ICubicWorldGenerator

// in (pre)init:
GameRegistry.registerWorldGenerator(hasCubicChunks ? new ModWorldGenerator() : new ModWorldGeneratorCubicChunks());
```

This would also make it possibe to have default implementation of IWorldGenerator in ICubicWorldGenerator for vanilla worlds.
And if the mod requires cubic chunks, and the code isn't only for compatibility - there would ne no hasCubicChunks check.